### ISSUE TITLE
cmd/tailscale/cli: [funnel/serve] changing node name leaves serve config that can't easily be cleared

### ISSUE THREAD

**Issue by shayne** [tailscale/tailscale#7086]
Reported from internal [Slack thread](https://tailscale.slack.com/archives/C021BPP12CR/p1674701411821929).

We currently record the node's name as part of the FQDN in the serve config. We should change that. If you change the node's name !) it keeps a record of the old name in the serve config, which is wrong; 2) the current `--remove` matches based on the current name, which means you need to do something like `$ echo {} | sudo tailscale serve set-raw` to remove them (which clears everything).

**Comment by vladsendrix** [comment]
I completely agree, this is something I have faced in the past. Funnel documentation only really scratches the surface and the cli is poorly implemented for the serve config.
